File: src/bindings/c/abm_forces.c
#include "abm_forces.h"
#include "string.h"

void calc_random_forces_c(int *cell_population,  int *force_field, double *force_magnitude);
void calc_saghian_chemo_forces_c(int *cell_population,  int *force_field, double *fradial, double *cradial, double *faxial, double *caxial);
void calc_saghian_cell_cell_c(int *cell_population,  int *force_field, double *r0, double *r1, double *a, double *b);

void calc_random_forces(int cell_population,  int force_field, double force_magnitude)
{
    calc_random_forces_c(&cell_population, &force_field, &force_magnitude);
}

void calc_saghian_chemo_forces(int cell_population,  int force_field, double fradial, double cradial, double faxial, double caxial)
{
	calc_saghian_chemo_forces_c(&cell_population, &force_field, &fradial, &cradial, &faxial, &caxial);
}

void calc_saghian_cell_cell(int cell_population,  int force_field, double r0, double r1, double a, double b)
{
	calc_saghian_cell_cell_c(&cell_population, &force_field, &r0, &r1, &a, &b);
}

File: src/bindings/c/abm_forces.f90
module abm_forces_c
implicit none
private

contains

!!!###################################################################################

subroutine calc_random_forces_c(cell_population, force_field, force_magnitude) bind(C, name="calc_random_forces_c")
use arrays, only: dp
use abm_forces, only: calc_random_forces
implicit none

integer,intent(in) :: cell_population
integer, intent(in) :: force_field
real(dp), intent(in) :: force_magnitude

#if defined _WIN32 && defined __INTEL_COMPILER
call so_calc_random_forces(cell_population, force_field, force_magnitude)
#else
call calc_random_forces(cell_population, force_field, force_magnitude)
#endif

end subroutine calc_random_forces_c

!###################################################################################

  subroutine calc_saghian_chemo_forces_c(cell_population, force_field, fradial, cradial, faxial, caxial) &
       bind(C, name="calc_saghian_chemo_forces_c")
    use arrays, only: dp
    use abm_forces, only: calc_saghian_chemo_forces
    implicit none

    integer,intent(in) :: cell_population
    integer, intent(in) :: force_field
    real(dp), intent(in) :: fradial,cradial,faxial,caxial

#if defined _WIN32 && defined __INTEL_COMPILER
    call so_calc_saghian_chemo_forces(cell_population, force_field,fradial,cradial,faxial,caxial)
#else
    call calc_saghian_chemo_forces(cell_population, force_field,fradial,cradial,faxial,caxial)
#endif

  end subroutine calc_saghian_chemo_forces_c

!###################################################################################

  subroutine calc_saghian_cell_cell_c(cell_population, force_field, r0,r1,a,b)bind(C, name="calc_saghian_cell_cell_c")
    use arrays, only: dp
    use abm_forces, only: calc_saghian_cell_cell
    implicit none

    integer,intent(in) :: cell_population
    integer, intent(in) :: force_field
    real(dp), intent(in) :: r0,r1,a,b

#if defined _WIN32 && defined __INTEL_COMPILER
    call so_calc_saghian_cell_cell(cell_population, force_field,r0,r1,a,b)
#else
    call calc_saghian_cell_cell(cell_population, force_field,r0,r1,a,b)
#endif

  end subroutine calc_saghian_cell_cell_c

end module abm_forces_c
